codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketHandshakeHandOverTest.java

```diff
@@ -184,6 +184,35 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
         }
     }
 
+    /**
+     * Tests a scenario when channel is closed while the handshake is in progress. Validates that the handshake
+     * future is notified in such cases.
+     */
+    @Test
+    public void testHandshakeFutureIsNotifiedOnChannelClose() throws Exception {
+        EmbeddedChannel clientChannel = createClientChannel(null);
+        EmbeddedChannel serverChannel = createServerChannel(null);
+
+        // Start handshake from client to server but don't complete the handshake for the purpose of this test.
+        transferAllDataWithMerge(clientChannel, serverChannel);
+
+        final WebSocketClientProtocolHandler clientWsHandler =
+                clientChannel.pipeline().get(WebSocketClientProtocolHandler.class);
+        final WebSocketClientProtocolHandshakeHandler clientWsHandshakeHandler =
+                clientChannel.pipeline().get(WebSocketClientProtocolHandshakeHandler.class);
+
+        final ChannelHandlerContext ctx = clientChannel.pipeline().context(WebSocketClientProtocolHandler.class);
+
+        // Close the channel while the handshake is in progress. This close could be initiated by the server by
+        // sending the CloseWebSocket frame. To reproduce the test scenario for this test case,
```

Comment:
Sorry maybe I miss something, how server can send close websocket frame if handshake is not complete yet.

Reply:
You are right. This is a bad example of a scenario for a channel close. I have changed the comment to remove this description. However, apart from this incorrect example, there can be number of cases when the channel might be closed before the handshake is complete. I have provide an example where SSL handshake fails below in the other comment.